dashboard/app/controllers/user_school_infos_controller.rb

```diff
@@ -0,0 +1,63 @@
+class UserSchoolInfosController < ApplicationController
+  #before_action :authenticate_user!
+
+  # PATCH /api/v1/users_school_infos/<id>/update_last_confirmation_date
+  def update_last_confirmation_date
+    user_school_info = UserSchoolInfo.find(params[:id])
+    result = user_school_info.update(last_confirmation_date: DateTime.now)
+    if result
+      head :no_content
+    else
+      render json: user_school_info.errors, status: :unprocessable_entity
+    end
+  end
+
+  # PATCH /api/v1/users_school_infos/<id>/update_end_date
+  def update_end_date
+    user_school_info = UserSchoolInfo.find(params[:id])
+    if user_school_info.update!(end_date: DateTime.now)
```

Comment:
is there a reason you're using a different error detection strategy here than you did in `update_last_confirmation_date`?

Reply:
I refactored the methods to have the same error detection.